tools/ami-cleanup/src/internal/cleanup.go

```diff
@@ -0,0 +1,239 @@
+		return trace.Wrap(err, "failed to load AWS credentials")
+	}
+
+	enabledRegions, err := ai.getEnabledRegions(ctx, ai.accountClientGenerator(&cfg))
+	if err != nil {
+		return trace.Wrap(err, "failed to get enabled regions")
+	}
+
+	if len(enabledRegions) == 0 {
+		return nil
+	}
+
+	totalSpaceRecovered := int32(0)
+	totalImagesDeleted := 0
+	// This would probably run much faster with channels/concurrency, but there isn't much
+	// need with how often this is expected to be ran
+	for _, enabledRegion := range enabledRegions {
+		spaceRecovered, imagesDeleted, err := ai.cleanupRegion(ctx, cfg, *enabledRegion.RegionName)
+		if err != nil {
+			return trace.Wrap(err, "failed to clean up images in region %q", *enabledRegion.RegionName)
+		}
+
+		totalSpaceRecovered += spaceRecovered
+		totalImagesDeleted += imagesDeleted
+	}
+
+	log.Printf("Deleted %d GiB of %d images across %d regions", totalSpaceRecovered, totalImagesDeleted, len(enabledRegions))
+	return nil
+}
+
+// Cleans up the images in the given region. Returns the total amount of space recovered, number of images deleted,
+// and any error.
+func (ai ApplicationInstance) cleanupRegion(ctx context.Context, cfg aws.Config, regionName string) (int32, int, error) {
+	// A new client must be created for each region
+	cfg.Region = regionName
+	ec2Client := ai.ec2ClientGenerator(&cfg)
+
+	log.Printf("Deleting images in %q", regionName)
+	devImages, err := ai.getDevImagesInRegion(ctx, ec2Client)
+	if err != nil {
+		return 0, 0, trace.Wrap(err, "failed to get a list of dev images for %q", regionName)
+	}
+
+	var totalSpaceRecovered int32
+	totalImagesDeleted := 0
+	totalSnapshotCount := 0
+	for _, devImage := range devImages {
+		imageSpace, err := ai.cleanupImageIfOld(ctx, ec2Client, devImage)
+		if err != nil {
+			return 0, 0, trace.Wrap(err, "failed to cleanup dev image %q", devImage.Name)
+		}
+
+		if imageSpace == 0 {
+			continue
+		}
+
+		totalSpaceRecovered += imageSpace
+		totalImagesDeleted++
+		totalSnapshotCount += len(devImage.BlockDeviceMappings)
+	}
+
+	return totalSpaceRecovered, totalImagesDeleted, nil
+}
+
+func (ai *ApplicationInstance) getEnabledRegions(ctx context.Context, client IAccountApi) ([]accountTypes.Region, error) {
+	return getAllWithPagination(
+		func(previousToken *string) (*string, []accountTypes.Region, error) {
+			results, err := client.ListRegions(ctx, &account.ListRegionsInput{
+				RegionOptStatusContains: []accountTypes.RegionOptStatus{
+					accountTypes.RegionOptStatusEnabled,
+					accountTypes.RegionOptStatusEnabledByDefault,
+				},
+				NextToken: previousToken,
+			})
+			if err != nil {
+				return nil, nil, trace.Wrap(err, "failed to request enabled regions")
+			}
+
+			return results.NextToken, results.Regions, nil
+		},
+	)
+}
+
+func (ai *ApplicationInstance) getDevImagesInRegion(ctx context.Context, client IEc2Api) ([]ec2Types.Image, error) {
+	// These are a weird language workaround for getting a pointer to a boolean literal
```

Comment:
I think this is what `aws.BoolValue(true)` is for. (And `aws.StringValue`)

Reply:
Thanks - I was not aware of these functions.